soroban-sdk/src/vec.rs

```diff
@@ -40,6 +40,32 @@ macro_rules! vec {
     };
 }
 
+macro_rules! impl_into_vec_for_tuple {
+    ( $($typ:ident $idx:tt)* ) => {
+        impl<$($typ),*> IntoVal<Env, Vec<RawVal>> for ($($typ,)*)
+        where
+            $($typ: IntoVal<Env, RawVal>),*
+        {
+            fn into_val(self, env: &Env) -> Vec<RawVal> {
+                vec![&env, $(self.$idx.into_val(env), )*]
+            }
+        }
```

Comment:
Can you add conversions from &TUPLE?
```suggestion
        }
        impl<E: Env, $($typ),*> IntoVal<E, RawVal> for &($($typ,)*)
         where
             $(for<'a> &'a $typ: IntoVal<E, RawVal>),*
         {
             fn into_val(self, env: &E) -> RawVal {
                 let env = env.clone();
                 let vec = env.vec_new($count.into());
                 $(let vec = env.vec_push(vec, (&self.$idx).into_val(&env));)*
                 vec.to_raw()
             }
         }
```

Reply:
Hmm the suggestion doesn't compile.

Reply:
Let's revert it, and I'll follow up with it.